src/Spd.Resource.Repository/BizLicApplication/BizLicApplicationRepository.cs

```diff
@@ -19,19 +20,29 @@ public BizLicApplicationRepository(IDynamicsContextFactory ctx, IMapper mapper)
 
     public async Task<BizLicApplicationResp> GetBizLicApplicationAsync(Guid licenceApplicationId, CancellationToken ct)
     {
-        spd_application? app = await _context.spd_applications
-            .Expand(a => a.spd_ServiceTypeId)
-            .Expand(a => a.spd_ApplicantId_account)
-            .Expand(a => a.spd_ApplicantId_contact)
-            .Expand(a => a.spd_application_spd_licencecategory)
-            .Expand(a => a.spd_application_spd_licence_manager)
-            .Expand(a => a.spd_CurrentExpiredLicenceId)
-            .Where(c => c.statecode != DynamicsConstants.StateCode_Inactive)
-            .Where(a => a.spd_applicationid == licenceApplicationId)
-            .FirstOrDefaultAsync(ct);
-
-        if (app == null) 
-            throw new ApiException(HttpStatusCode.NotFound);
+        spd_application? app;
+        try
+        {
+            app = await _context.spd_applications
+                .Expand(a => a.spd_ServiceTypeId)
+                .Expand(a => a.spd_ApplicantId_account)
+                .Expand(a => a.spd_ApplicantId_contact)
+                .Expand(a => a.spd_application_spd_licencecategory)
+                .Expand(a => a.spd_application_spd_licence_manager)
+                .Expand(a => a.spd_CurrentExpiredLicenceId)
+                .Where(a => a.spd_applicationid == licenceApplicationId)
+                .FirstOrDefaultAsync(ct);
+        }
+        catch (DataServiceQueryException ex)
+        {
+            if (ex.Response.StatusCode == 404)
+                app = null;
+            else
+                throw;
+        }
+
+        if (app == null)
```

Comment:
This condition looks redundant, considering that we already have a `catch` statement. Is this correct?

Reply:
test